grakn-graql/src/main/java/ai/grakn/graql/internal/query/match/MatchQueryInfer.java

```diff
@@ -49,7 +49,7 @@
     public Stream<Answer> stream(Optional<GraknGraph> optionalGraph) {
         GraknGraph graph = optionalOr(optionalGraph, inner.getGraph()).orElseThrow(GraqlQueryException::noGraph);
 
-        if (!ReasonerUtils.hasRules(graph)) return inner.stream(optionalGraph);
+        if (!new RuleGraph(graph).hasRules()) return inner.stream(optionalGraph);
```

Comment:
[*] - This could potentially be `RuleUtil.hasRules(graph)`, no object allocation needed